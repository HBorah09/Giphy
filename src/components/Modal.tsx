import { FC } from 'react';
import { Modal as BootstrapModal, Button } from 'react-bootstrap';
import { Gif } from './Giphys';

interface Props {
    show: boolean,
    handleClose: () => void,
    data: Gif
}

const Modal: FC<Props> = ({ show, handleClose, data }) => {

    return(
        <BootstrapModal show={show} keyboard onHide={handleClose}>
            <BootstrapModal.Header closeButton>
            <BootstrapModal.Title>{data.title}</BootstrapModal.Title>
            </BootstrapModal.Header>
            <BootstrapModal.Body>
                <img src={data.images.fixed_height.url} alt={data.title} style={{width: '100%'}}  />
            </BootstrapModal.Body>
            <BootstrapModal.Footer>
            <Button variant="secondary" onClick={handleClose}>
                Close
            </Button>
            </BootstrapModal.Footer>
        </BootstrapModal>
    )
}

export default Modal;