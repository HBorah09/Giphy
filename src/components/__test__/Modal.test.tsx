import { shallow } from 'enzyme'
import Modal from '../Modal';

describe('<Modal />', () => {

    const props = {
        show: true, 
        handleClose : () => jest.fn(),
        data: {
            id: '123',
            title: 'funny gif',
            images: {
                fixed_height: {
                    url: ''
                }, 
                original_still: {
                    url: ''
                }
            },
        }
    }

    it('renders without crashing', () => {
        const wrapper = shallow(<Modal { ...props } />);
        expect(wrapper).toMatchSnapshot();
    });
})