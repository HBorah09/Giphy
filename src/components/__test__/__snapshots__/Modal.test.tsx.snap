// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders without crashing 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      funny gif
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <img
      alt="funny gif"
      src=""
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
